decryptor/postgresql/pg_general_decryptor.go

```diff
@@ -476,15 +397,176 @@ func (decryptor *PgDecryptor) MatchZoneInBlock(block []byte) {
 
 // GetTagBeginLength returns begin tag length, depends on decryptor type
 func (decryptor *PgDecryptor) GetTagBeginLength() int {
-	return decryptor.pgDecryptor.GetTagBeginLength()
+	return decryptor.binaryDecryptor.GetTagBeginLength()
 }
 
 // GetZoneIDLength returns begin tag length, depends on decryptor type
 func (decryptor *PgDecryptor) GetZoneIDLength() int {
-	return decryptor.pgDecryptor.GetTagBeginLength()
+	return decryptor.binaryDecryptor.GetTagBeginLength()
 }
 
 // SetDataProcessor replace current with new processor
 func (decryptor *PgDecryptor) SetDataProcessor(processor base.DataProcessor) {
 	decryptor.dataProcessor = processor
 }
+
+// OnColumn handler which process column data on db response and try to decrypt/detect poison record
+func (decryptor *PgDecryptor) OnColumn(ctx context.Context, data []byte) (context.Context, []byte, error) {
+	logger := logging.GetLoggerFromContext(ctx)
+	span := trace.FromContext(ctx)
+	// try to skip small piece of data that can't be valuable for us
+	if (decryptor.IsWithZone() && len(data) >= zone.ZoneIDBlockLength) || len(data) >= base.GetMinAcraStructLength() {
+		decryptor.Reset()
+		span.AddAttributes(trace.BoolAttribute("decryption", true))
+
+		// Zone anyway should be passed as whole block
+		// so try to match before any operations if we process with ZoneMode on
+		if decryptor.IsWithZone() && !decryptor.IsMatchedZone() {
```

Comment:
_random suggestion, don't shout @ me_

maybe it makes sense to extract this `if` into separate method `matchZone(decryptor, data, logger)`? To improve readability of this code